modules/schema_validation.lua

```diff
@@ -325,28 +331,19 @@ local function compare(schema, function_id, msgType, user_data, mandatory_check)
   return schemaCompare(xml_schema,user_data)
 end
 
-function module.validate_hmi_request(function_id,user_data, mandatory_check)
-  return compare(module.HMI, function_id, 'request', user_data, mandatory_check)
-end
 
-function module.validate_mobile_request(function_id,user_data, mandatory_check)
-  return compare(module.MOBILE,function_id, 'request',user_data, mandatory_check)
-end
-
-function module.validate_hmi_response(function_id,user_data, mandatory_check)
-  return compare(module.HMI,function_id, 'response',user_data, mandatory_check)
-end
 
-function module.validate_mobile_response(function_id,user_data, mandatory_check)
-  return compare(module.MOBILE,function_id, 'response',user_data, mandatory_check)
-end
 
-function module.validate_mobile_notification(function_id,user_data, mandatory_check)
-  return compare(module.MOBILE,function_id, 'notification',user_data, mandatory_check)
-end
+function module.mt.__index:Validate(function_id, function_type, user_data)
+  local result = true
+  local errorMessage = {}
 
-function module.validate_hmi_notification(function_id,user_data, mandatory_check)
-  return compare(module.HMI,function_id, 'notification',user_data, mandatory_check)
-end
+  if (api_type == "MOBILE") then
+    result, errorMessage = compare(module.MOBILE,function_id, function_type, user_data, mandatory_check)    
+  else
+    result, errorMessage = compare(module.HMI,function_id, function_type, user_data, mandatory_check)  
+  end
+    return result, errorMessage
+  end
```

Comment:
@pestOO Final commit has no if statement, this is the intermediate version before schema refactoring